modules/aggregation-layers/src/aggregation-layer-v9/aggregator.ts

```diff
@@ -0,0 +1,65 @@
+import type {Attribute, BinaryAttribute} from '@deck.gl/core';
+
+export type AggregationOperation = 'SUM' | 'MEAN' | 'MIN' | 'MAX';
+
+export type AggregationProps = {
+  /** Number of data points */
+  pointCount: number;
+  /** The input data */
+  attributes: {[id: string]: Attribute};
+  /** How to aggregate getWeights, defined for each channel */
+  operations: AggregationOperation[];
+  /** Additional options to control bin sorting, e.g. bin size */
+  binOptions: Record<string, number | number[]>;
+};
+
+/**
+ * _Aggregation_ is a 2-step process:
+ * 1. Sort: Group a collection of _data points_ by some property into _bins_.
+ * 2. Aggregate: for each _bin_, calculate one or more metrics (_channels_) from all its members.
+ *
+ * An implementation of the _Aggregator_ interface takes the following inputs:
+ * - The number of data points
+ * - The group that each data point belongs to, by mapping each data point to a _binId_ (integer or array of integers)
+ * - The value(s) to aggregate, by mapping each data point in each channel to one _weight_
+ * - The method (_aggregationOperation_) to reduce a list of _weights_ to one number, such as SUM
```

Comment:
Minor – I felt like the term "weights" was a bit misleading in the previous iteration of aggregation... For me it implies that they are coefficients for something, rather than a bunch of numbers we're going to sum or average. I think "values" or "inputs" might work just as well here without that confusion?